docs/content/RelativeTime.mdx

```diff
@@ -0,0 +1,141 @@
+---
+componentId: relative-time
+title: RelativeTime
+status: Alpha
+source: https://github.com/primer/react/blob/main/src/RelativeTime.tsx
+---
+
+```js
+import {RelativeTime} from '@primer/react'
+```
+
+Formats a timestamp as a localized string or as relative text that auto-updates in the user's browser.
+
+## Examples
+
+### Default
+
+```jsx live
+<>
+  <RelativeTime datetime="2014-04-01T16:30:00-08:00" />
+  <br/>
+  <RelativeTime date={new Date('2014-04-01T16:30:00-08:00')} format="micro" tense="past"/>
+  <br/>
+  <RelativeTime datetime="2014-04-01T16:30:00-08:00" format="elapsed"/>
+</>
+```
+
+## Props
+
+### RelativeTime
+
+<PropsTable>
+  <PropsTableRow
+    name="datetime"
+    type="string"
+    description="An ISO8601 time represeting the time"
+  />
+  <PropsTableRow
+    name="date"
+    type="string"
+    description="Instead of passing the `datetime`, it is possible to pass a full `Date` object into the `date` prop"
+  />
+  <PropsTableRow
+    name="format"
+    type="'auto' | 'micro' | 'elapsed' | string"
+    defaultValue="'auto'"
+    description="The format with which to display the time"
+  />
+  <PropsTableRow
+    name="tense"
+    type="'auto' | 'past' | 'future'"
+    defaultValue="'auto'"
+    description="The tense to use when displaying a relative time"
+  />
+  <PropsTableRow
+    name="precision"
+    type="'year' | 'month' | 'day' | 'hour' | 'minute' | 'second'"
+    defaultValue="'second'"
+    description="The precision to use when displaying an elapsed time"
+  />
+  <PropsTableRow
+    name="threshold"
+    type="string"
+    defaultValue="'P30D'"
+    description="The threshold (ISO8601 duration) to display relative dates within"
```

Comment:
Looking at the [design guidelines for relative time](https://primer-437c43db97-26441320.drafts.github.io/components/relative-time), are there options here that we should bake into the component instead of making it configurable?

Reply:
I believe RelativeTime _is_ opinionated; it has a good set of sensible defaults and there should be very little need to customise it. However date-times are a sensitive topic that are very dependent on context (I've made a PR https://github.com/primer/design/pull/331 which goes into some explanation of the principles here).

`threshold` is configurable because sometimes we want a no threshold (i.e. `PT0S`) and sometimes a much shorter threshold (e.g. `P1D`). The default of `P30D` works in 99% of use cases. From the guidelines PR:

> Some dates are especially sensitive and always should be displayed in a precise date-time format, such as the creation or expiration of certificates and keys. When a user needs to take action before a particular time (such as an expiry date) then the precise date time should be displayed

`precision` is also configurable to allow us to elide unimportant information in certain contexts. You care how many seconds your GitHub Action is taking, less so about how long it's been since a PAT was last used.

The `format` prop supporting custom formatting is definitely an escape hatch, and it is something I'd like to deprecate (bearing in mind this component has existed for a long time)! We _do_ want to move to supporting only recommended options which is why `second`, `weekday`, etc all exist. They will allow for configuration of dates and times in a way which allows flexibility while ensuring good practices are followed.